### PR TITLE
add configuration-as-code plugin to Bill of Material

### PR DESCRIPTION
**Pull request by casz**
cc @jglick @oleg-nenashev hopefully I did this right 😅

### CODE REVIEW

File: sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+public class ConfigurationAsCodeTest {
+
+    @ClassRule
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public static JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    public void smokes() {
+        assertThat(Jenkins.get().getSystemMessage(), is("Hello World"));
```
[the remark this answers is not in the document]
Comment (casz) [reply sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java]: There is nothing really interesting to configure in JCasC 😆

**Comment by casz** [comment]
Should I go ahead and bump git in this PR?

File: sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (casz) [reply sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java]: https://github.com/jenkinsci/bom/commit/91d7c3ee262060c20f2f721d3b0d93ca59261d4e hmm
@MarkEWaite 😅

[the remark this answers is not in the document]
Comment (casz) [reply sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java]: Just wanna state BOM and PCT is pretty awesome! Kudos 😍

**Comment by casz** [comment]
Same issue. They both need updating of JCasC just as Git did.

**Comment by casz** [comment]
@MRamonLeon are you up for fixing them? 😅

**Comment by MRamonLeon** [comment]
Sorry @casz, I don't have enough bandwidth right now, maybe next week.
For **ssh-slaves**, looks like JCasC is not executing after the restart, there isn't any log of JCasC.

For **ssh-credentials**, maybe a security issue has been solved and the log doesn't contain the plain password anymore and the test needs to be fixed:
`The log should have 'passphrase = passphrase-of-userid2'`

`0.690 [id=693]	FINE	i.j.p.c.i.c.DataBoundConfigurator#tryConstructor: Setting class com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey. passphrase = ****`

Please take into account that the `RoundTripAbstractTest` test I did is just a proxy, a utility to develop JCasC complete tests quickly, so it's going to be in the middle of every failure, but don't shoot the messenger :sweat_smile:

**Comment by casz** [comment]
@MRamonLeon I'll see if I have bandwidth for it 😅

**Comment by casz** [comment]
Looking into ssh-credentials 😓

**Comment by timja** [comment]
> Looking into ssh-credentials 😓

hah, I just created a branch on that repo, it'll require this to be merged:
https://github.com/jenkinsci/configuration-as-code-plugin/pull/1049

**Comment by casz** [comment]
Okay I'll look into the other one @timja so looking into `ssh-slaves` 😓

**Comment by casz** [comment]
I see someone else did my work: https://github.com/jenkinsci/ssh-slaves-plugin/pull/147 thanks @fcojfernandez

File: sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (casz) [reply sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java]: Should I remove the test?

[the remark this answers is not in the document]
Comment (jglick) [reply sample-plugin/src/test/java/io/jenkins/tools/bom/sample/ConfigurationAsCodeTest.java]: It is fine.

**Comment by casz** [comment]
@jglick would you prefer dependabot updating the deps or are you okay with me updating the deps in this PR?

**Comment by jglick** [comment]
If you have some dependencies related to this change that you see could be updated, by all means update them.

File: bom/pom.xml [hunk 2]
```diff
@@ -148,7 +160,7 @@
             <dependency>
                 <groupId>org.jenkins-ci.plugins</groupId>
                 <artifactId>git-client</artifactId>
-                <version>2.8.4</version>
+                <version>3.0.0-beta11</version>
```
Comment (timja) [inline bom/pom.xml]:
```suggestion
                <version>2.8.5</version>
```

Comment (casz) [reply bom/pom.xml]: 2.8.5 includes the broken tests, so I think @MarkEWaite we need a release for this, or incremental for 2.8.6, or a snapshot will do too just to confirm

Comment (timja) [reply bom/pom.xml]: ah 😢

Comment (jglick) [reply bom/pom.xml]: Superseded by #102 I think?

Comment (timja) [reply bom/pom.xml]: yes

**Comment by casz** [comment]
🙏 for green

**Comment by casz** [comment]
@jglick not sure the CI is so read 😕❓

**Comment by jglick** [comment]
> the CI is so red

Sigh, yes, seems like Remoting problems on ci.jenkins.io again.

**Comment by timja** [comment]
> > the CI is so red
> 
> Sigh, yes, seems like Remoting problems on ci.jenkins.io again.

I can't even load this CI job =/, it's been loading for +20 mins

**Comment by timja** [comment]
🙏

**Comment by timja** [comment]
Whoop green, cc @jglick

**Comment by casz** [comment]
This would be great to get merged!

**Pull request merged**